x/incentives/keeper/gauge.go

```diff
@@ -281,28 +275,33 @@ func (k Keeper) CreateGroupGauge(ctx sdk.Context, coins sdk.Coins, numEpochPaidO
 	return nextGaugeId, nil
 }
 
-// AddToGaugeRewardsFromGauge transfer coins from groupGaugeId to InternalGaugeId.
-// Prior to calling this function, we make sure that the internalGaugeId is linked with the associated groupGaugeId.
-// Note: we donot have to bankSend for this gauge transfer because all the available incentive has already been bank sent
-// when we create Group Gauge. Now we are just allocating funds from groupGauge to internalGauge.
-func (k Keeper) AddToGaugeRewardsFromGauge(ctx sdk.Context, groupGaugeId uint64, coins sdk.Coins, internalGaugeId uint64) error {
-	// check if the internalGaugeId is present in groupGauge
-	groupGaugeObj, err := k.GetGroupGaugeById(ctx, groupGaugeId)
-	if err != nil {
-		return err
-	}
-	found := false
-	for _, val := range groupGaugeObj.InternalIds {
-		if val == internalGaugeId {
-			found = true
-			break
+// nolint: unused
+// getWeightBySplittingPolicy takes in a pool ID, splitting policy, and current weight, and returns the updated weight (should be positive)
+func (k Keeper) getWeightBySplittingPolicy(ctx sdk.Context, poolId uint64, splittingPolicy types.SplittingPolicy, currentWeight osmomath.Int) (osmomath.Int, error) {
+	switch splittingPolicy {
+	case types.Evenly:
+		return osmomath.OneInt(), nil
+	case types.Volume:
+		// Set the weight to diff between the current volume of the pool
+		totalVolume := k.pmk.GetOsmoVolumeForPool(ctx, poolId)
+
+		if totalVolume.LT(currentWeight) {
+			return osmomath.ZeroInt(), types.CumulativeVolumeDecreasedError{PoolId: poolId, PreviousVolume: currentWeight, NewVolume: totalVolume}
```

Comment:
is returning zero here as opposed to empty struct intentional? Empty struct is generally preferred with errors unless specific reason for doing otherwise.

Reply:
In most of these cases returning zero is intentional due to quiet error handling at higher levels, but in this case an empty struct should have been returned. The function was unused and vestigial from an earlier (scrapped) architecture and is now removed, but nice catch regardless.